src/pk-main.c

```diff
@@ -182,7 +182,16 @@ main (int argc, char *argv[])
 	syslog (LOG_DAEMON | LOG_DEBUG, "daemon start");
 
 	/* after how long do we timeout? */
-	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", NULL);
+	exit_idle_time = g_key_file_get_integer (conf, "Daemon", "ShutdownTimeout", &error);
+	/* THIS COMMENT IS A TSUNAMI STONE
+         * The automatic shutdown timeout is necessary to avoid very high
+         * memory usage with the dnf backend.  If you want to remove this
```

Comment:
style: Please use tabs and a single space, not just spaces.

Reply:
Done, after a brief detour to set up a good `listchars` in my vimrc.